userapp/src/components/table/HelixTable.js

```diff
@@ -101,12 +64,39 @@ const HelixTable = ({
     setOrderBy(property)
   }
 
+  // searchFilter contains a func that store filter query search upon user input
+  const [searchFilter, setSearchFilter] = useState({ search: (rows, columns) => { return rows }})
+  
+  /**
+   * @param {object} event the event object contains user input
+   * Pass the user query input to searchFilter and it store which object matches the query 
+   */
+  const onSearch = (event) => {
+    const { value } = event.target
+    setSearchFilter({ search: (rows, columns) => {
+        if (value === '') return rows
+        else 
+          return rows.filter((row) => 
+            (columns.filter((column) => 
+              row[column.ID]
+              .toLowerCase()
+              .includes(value.toLowerCase()))
+              .length > 0 
+              ? true
+              : false
+            )
```

Comment:
should I make a helper function that return filtered array of column object that matches row data? 
`(columns.filter((column) => 
              row[column.ID]
              .toLowerCase()
              .includes(value.toLowerCase()))`

row1: [ {Label: FirstName, ID: First Name}, {Label: Last Name, ID: LastName} ]
row2:[ ]
row3: [ {Label: FirstName, ID: First Name} ]